### PR TITLE
Add CNI config 1.0.0 support.

### PR DESCRIPTION
**Pull request by aznashwan**
This PR updates the `containernetwork/cni` dependency to 1.1.2 and adds support for reading CNI configs of up to 1.0.0.

Note that, although the "CHECK" verb had an entry point added for it it, it is currently no-op.

### CODE REVIEW

**Comment by aznashwan** [comment]
I am unsure why the 2022 runner is [failing to find the main interface](https://github.com/microsoft/windows-container-networking/actions/runs/7181999624/job/19557543016?pr=101#step:8:175), as I have managed to successfully run it 5 times on my fork. ([104](https://github.com/aznashwan/windows-container-networking/actions/runs/7182935686), [105](https://github.com/aznashwan/windows-container-networking/actions/runs/7182935686), [106](https://github.com/aznashwan/windows-container-networking/actions/runs/7183271048), [107](https://github.com/aznashwan/windows-container-networking/actions/runs/7183271714), and [108](https://github.com/aznashwan/windows-container-networking/actions/runs/7183272252))

@debj1t could you please re-run the CI a couple more times so we can see whether it's an issue on this repo or transient to the runners.

File: common/core/network.go [hunk 1]
```diff
@@ -484,3 +484,13 @@ func (plugin *netPlugin) Delete(args *cniSkel.CmdArgs) error {
...
 	return nil
 }
+
+// Check handles CNI CHECK commands.
+// args.ContainerID - ID of the container for which network endpoint is to be checked.
+// args.Netns - Network Namespace Id (required).
+// args.IfName - Interface Name specifies the interface the network should bind to (ex: Ethernet).
+// args.Path - Location of the config file.
+func (plugin *netPlugin) Check(args *cniSkel.CmdArgs) error {
+	logrus.Warnf("[cni-net] CHECK is currently NOT implemented! Called with args: %v", args)
```
Comment (debj1t) [inline common/core/network.go:494]: Can we create an issue for pending CHECK-verb/disableCheck-field implementation just to have it documented and track later if required?

File: cni/cni.go [hunk 2]
```diff
@@ -14,20 +14,21 @@ import (
...
+	CmdAdd   = "ADD"
+	CmdDel   = "DEL"
+	CmdCheck = "CHECK"
 
 	Internal = "internal"
 )
 
 // Supported CNI versions.
-var VersionsSupported = []string{"0.2.0", "0.3.0"}
+var VersionsSupported = []string{"0.2.0", "0.3.0", "1.0.0"}
```
Comment (debj1t) [inline cni/cni.go:31]: We can also add 0.4.0 here? Any additional changes required for 0.4.0?

**Review by debj1t — CHANGES_REQUESTED** [review]

File: cni/cni.go (hunk 2 above)
Comment (aznashwan) [reply cni/cni.go]: Good catch! I have just added `0.4.0` to the list as well!

> Any additional changes required for 0.4.0?

As per [update guidelines for 0.4.0](https://github.com/containernetworking/cni/blob/main/Documentation/spec-upgrades.md#changes-in-cni-v04), all that is needed from the plugin end was handling `CHECK`. (I had initially forgotten to [hook it into the CniSkel](https://github.com/microsoft/windows-container-networking/pull/101/files#diff-62698665e85b9e578006a98b0335b734e0ea9bf88c2704cb237c0fc76014a61eR48) but have fixed that)

The [update guidelines for 1.0.0](https://github.com/containernetworking/cni/blob/main/Documentation/spec-upgrades.md#libcni-changes-in-cni-v10) only mentions [switching away from `types/current` to `types/XYZ`](https://github.com/microsoft/windows-container-networking/pull/101/files#diff-8a47859b51210f40efd508827dea21a220fc7c0d7fee3f8d2a87ec38220a26baL17).

**Review by debj1t — APPROVED** [review]
Thanks, for all the work!

**Comment by nicolaerosia** [comment]
@debj1t is there a way I can test this out? I would like to get an archive with the artefacts

**Comment by aznashwan** [comment]
@nicolaerosia back in December I had written [this workflow running containerd's integration tests against the plugins](https://github.com/aznashwan/windows-container-networking/blob/containerd-integration/.github/workflows/containerd-integration.yml).

Looking at [one of my latest runs](https://github.com/aznashwan/windows-container-networking/actions/runs/6850896195), the `nat` tests all ran fine and the `sdnbridge` ones were only failing at the PortMapping tests. (by default, the `sdnbridge` plugin does NOT create a gateway endpoint to allow the containers to reach the host)
`sdnoverlay` was blocked at initial network creation, however.

NOTE: there is a slight [issue with the naming of the plugins not coinciding with the HNS network types](https://github.com/microsoft/windows-container-networking/issues/57), so you will need to also include [this other PR](https://github.com/microsoft/windows-container-networking/pull/99) in your testing. (it was the final blocker which prevented full integration with containerd before I left the project)

Hope this helped a bit, godspeed!

**Comment by slonopotamus** [comment]
Could you please make a release with these changes?

**Pull request merged**